### PR TITLE
Token verifier

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Changes proposed in this pull request:

- Add token verifier application. It uses oidc package from github.com/kyma-project/test-infra repository.
- Application uses a cobra command framework for cli syntax.
- Application uses OIDC discovery to get pubilc keys to verify token.
- Verified token claims are written as json to the file provided in flag.
- Application can read a raw oidc token from AUTHORIZATION environment variable.
- Logging uses a zap logger.
- Code contains comments related to next feature using cached public keys.

**Related issue(s)**
See: https://github.com/orgs/kyma-project/projects/22/views/70?pane=issue&itemId=56193547

### CODE REVIEW

**Comment by dekiel** [comment]
This PR can be merged after #10607 which provides a oidc library used by token verifier application.

File: cmd/oidc-token-verifier/main.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+// It sets the environment variable specified by --new-public-keys-var-name to true to indicate that new public keys are fetched.
+// func (opts *options) savePublicKeysFromRemote(issuer string) error {
+//
+// }
+
+// setAdoEnvVar sets the Azure DevOps pipeline environment variable to true.
+// Environment variable name is specified by --new-public-keys-var-name flag.
+// func (opts *options) setAdoEnvVar() error {
+//
+// }
```
Comment (Sawthis) [inline cmd/oidc-token-verifier/main.go:196]: should these comments be in this place? They look like leftovers for commented functions.

**Review by Sawthis — CHANGES_REQUESTED** [review]
`cmd/oidc-token-verifier/main.go:8:2: no required module provides package github.com/kyma-project/test-infra/pkg/oidc; to add it:
	go get github.com/kyma-project/test-infra/pkg/oidc`

File: cmd/oidc-token-verifier/main.go [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+	rootCmd.PersistentFlags().StringVarP(&opts.clientID, "client-id", "c", "image-builder", "OIDC token client ID")
+	rootCmd.PersistentFlags().StringVarP(&opts.publicKeyPath, "public-key-path", "p", "", "Path to the cached public keys directory")
+	rootCmd.PersistentFlags().BoolVarP(&opts.debug, "debug", "d", false, "Enable debug mode")
+	return rootCmd
+}
+
+func NewClaimsCmd() *cobra.Command {
+	claimsCmd := &cobra.Command{
+		Use:   "claims",
+		Short: "Work with OIDC claims",
```
[the remark this answers is not in the document]
Comment (dekiel) [reply cmd/oidc-token-verifier/main.go:57]: This command act as group name for other commands which works with OIDC claims. At present there is no specific action assigned to this command directly, but it has child command `extract`. Thus command will be `oidc claims extract`.

File: cmd/oidc-token-verifier/main.go (hunk 1 above)
Comment (dekiel) [reply cmd/oidc-token-verifier/main.go:196]: Yes, they should be in a file. These are comment for feature of using cached public keys.

File: cmd/oidc-token-verifier/main.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
```
Comment (Sawthis) [inline cmd/oidc-token-verifier/main.go:57]:
```suggestion
		Short: "OIDC claims related commands",
```

**Review by Sawthis — CHANGES_REQUESTED** [review]

**Review by Sawthis — CHANGES_REQUESTED** [review]
linter is complaining
`cmd/oidc-token-verifier/main.go:72:14: unused-parameter: parameter 'cmd' seems to be unused, consider removing or renaming it as _ (revive)
		RunE: func(cmd *cobra.Command, args []string) error {
	`

**Review by Sawthis — APPROVED** [review]

**Pull request merged**